client.go

```diff
@@ -270,11 +270,10 @@ func (c *Client) setDefaults() {
 	c.confirmMode = true
 }
 
-// runForever will connect amqp, setup all the amqp channels, run the publisher
-// and run the replies consumer. The method will also automatically restart
-// the setup if the underlying connection or socket isn't gracefully closed.
-// This will also block until the client is gracefully stopped.
-func (c *Client) runForever() {
+// RunForever will start the client and run it forever, reconnecting
+// automatically when needed. Note that this will return as soon as the setup
+// is started, to wait for the setup to be completed use [Client.OnStarted].
+func (c *Client) RunForever() {
```

Comment:
If this will be public, how about `Connect`? It feels weird to call `Run` on a client, to the user it's not obvious why it needs to be "running". However I think connecting is more common with clients such as SQL clients etc.

Reply:
Yes! That is a much better name!